### PR TITLE
Update gtest used to v1.15.2

### PR DESCRIPTION
**Pull request by mcbarton**
This PR will attempt to update the version of gtest we are using to version 1.15.2 .
Fixes https://github.com/compiler-research/CppInterOp/issues/295

### CODE REVIEW

File: cmake/modules/GoogleTest.cmake [hunk 1]
```diff
@@ -17,14 +17,15 @@ if(WIN32)
     BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config $<CONFIG>)
 elseif(APPLE)
   set(EXTRA_GTEST_OPTS -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT})
+  string(REPLACE "-Wc++17-attribute-extensions" "" CMAKE_CXX_FLAGS "${CMAKE_Cxx_FLAGS}")
```
Comment (vgvassilev) [inline cmake/modules/GoogleTest.cmake:20]:
```suggestion
  string(REPLACE "-Wc++17-attribute-extensions" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Comment by vgvassilev** [comment]
Please squash on merge.

**Comment by mcbarton** [comment]
@vgvassilev This PR needs another review, as I had to take a different approach after your last review revealed an error in my previous approach.

File: cmake/modules/GoogleTest.cmake [hunk 2]
```diff
@@ -16,15 +16,20 @@ if(WIN32)
     -Dgtest_force_shared_crt=ON
     BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config $<CONFIG>)
 elseif(APPLE)
+  #FIXME: Temporarily remove -pedantic from CMAKE_CXX_FLAGS to avoid issues with building
+  #       latest version of gtest on MacOS (use of -Wc++17-attribute-extensions causes 
+  #       issues due to [[maybe_unused]] in gtest)
   set(EXTRA_GTEST_OPTS -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT})
+  set(CMAKE_CXX_FLAGS_TEMP_COPY ${CMAKE_CXX_FLAGS})
+  string(REPLACE "-pedantic" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```
Comment (vgvassilev) [inline cmake/modules/GoogleTest.cmake:24]: I am wondering if we need to propagate `CMAKE_CXX_COMPILER` to the `CMAKE_ARGS`. Alternatively in `CMAKE_ARGS` we can append `-no-pedantic` to disable the default mode.

Comment (vgvassilev) [reply cmake/modules/GoogleTest.cmake:24]: ping.

**Comment by mcbarton** [comment]
@vgvassilev Rather than add `-no-pedantic` in the osx build I just found where `pedantic` was being applied in the osx builds and removed. Is the PR good now?

**Pull request merged**